Makefile

```diff
@@ -543,6 +544,61 @@ proto:
 		--gogo_out=plugins=grpc,$(PROTO_GOGO_MAPPINGS):$(PWD)/proto-gen/zipkin \
 		idl/proto/zipkin.proto
 
+	$(PROTOC) \
+		$(PROTO_INCLUDES) \
+		--gogo_out=plugins=grpc,paths=source_relative,$(PROTO_GOGO_MAPPINGS):$(PWD)/proto-gen/otel \
+		$(PROTO_INTERMEDIATE_DIR)/common/v1/common.proto
+	$(PROTOC) \
+		$(PROTO_INCLUDES) \
+		--gogo_out=plugins=grpc,paths=source_relative,$(PROTO_GOGO_MAPPINGS):$(PWD)/proto-gen/otel \
+		$(PROTO_INTERMEDIATE_DIR)/resource/v1/resource.proto
+	$(PROTOC) \
+		$(PROTO_INCLUDES) \
+		--gogo_out=plugins=grpc,paths=source_relative,$(PROTO_GOGO_MAPPINGS):$(PWD)/proto-gen/otel \
+		$(PROTO_INTERMEDIATE_DIR)/trace/v1/trace.proto
+
+	# Target  proto-prepare-otel modifies OTEL proto to use import path jaeger.proto.*
+	# The modification is needed because OTEL collector already uses opentelemetry.proto.*
+	# and two complied protobuf types cannot have the same import path. The root cause is that the compiled OTLP
+	# in the collector is in private package, hence it cannot be used in Jaeger.
```

Comment:
Isn't there a ticket in OTEL to export that package? It do we not want to use it?

Reply:
I didn't see any open issue. It has been private from the get-go.

Reply:
We have a similar process in Tempo which is here:
https://github.com/grafana/tempo/blob/main/Makefile#L129

We have also since forked the collector and written a script to just rename `internal` => `external` so we can have access to more of the otel guts. We will probably move to using this soon instead of our current setup.
https://github.com/grafana/opentelemetry-collector/tree/0.29-grafana

> I didn't see any open issue. It has been private from the get-go.
It used to be exposed b/c Tempo vendored it initially. They moved it internal and we asked about keeping it exposed. They opted to keep it internal to protect themselves from people forming a dependency on it.

Reply:
I didn't see it being open, but good to know. Perhaps they have reasons to keep it intenal. They even spent a lot of time on the (ugly) pdata package to make that happen.